crossfit/backend/torch/loader.py

```diff
@@ -138,8 +159,17 @@ def __next__(self):
                     for key, val in self.tensor_dict.items()
                     if key not in self.to_ignore
                 }
-                clip_len = min(max(_tokens[start], _tokens[end - 1]), self.model.max_seq_length())
-                batch = {key: val[:, :clip_len] for key, val in batch.items()}
+                # TODO: Fix max_length
+                if self.max_seq_len is None:
```

Comment:
What is left to do here? Also, shouldn't `self.max_seq_len` always be set here since you set it in the constructor, or am I missing an edgecase?

Reply:
It should be but was not being set and i could not figure out why .

Reply:
I figured it out. You are calling `super().__init__(frame, initial_batch_size, progress_bar=progress_bar)`on line 130 which resets it to None after you set it on line 122. I'd just remove the assignment on line 122 and pass `self.model.max_seq_length()` into the super constructor

Reply:
That was it, fixed it.